ui/widgets.go

```diff
@@ -148,3 +169,49 @@ func mustIcon(ic *decredmaterial.Icon, err error) *decredmaterial.Icon {
 	}
 	return ic
 }
+
+func (win *Window) initTransactionsWidgets(transactions []wallet.TransactionInfo) {
+	win.combined.transactions = nil
+
+	for i := 0; i < len(transactions); i++ {
+		iconStatus, _ := decredmaterial.NewIcon(icons.ToggleRadioButtonUnchecked)
+		if transactions[i].Status == "confirmed" {
+			iconStatus, _ = decredmaterial.NewIcon(icons.ActionCheckCircle)
+			iconStatus.Color = win.theme.Color.Success
+		}
+
+		var iconDirection struct {
+			icon            string
+			backgroundColor color.RGBA
+			innerColor      color.RGBA
+		}
+
+		switch transactions[i].Direction {
+		case "Sent":
+			iconDirection.icon = "-"
+			iconDirection.backgroundColor = win.theme.Color.Danger
+			iconDirection.innerColor = color.RGBA{254, 209, 198, 255}
+		case "Received", "Yourself":
+			iconDirection.icon = "+"
+			iconDirection.backgroundColor = win.theme.Color.Success
+			iconDirection.innerColor = color.RGBA{198, 236, 203, 255}
+		}
+
+		transaction := struct {
+			iconStatus    *decredmaterial.Icon
+			iconDirection struct {
+				icon            string
+				backgroundColor color.RGBA
+				innerColor      color.RGBA
+			}
+			data    interface{}
+			gesture *gesture.Click
+		}{
+			iconStatus,
+			iconDirection,
+			&transactions[i],
+			&gesture.Click{},
+		}
+		win.combined.transactions = append(win.combined.transactions, transaction)
+	}
+}
```

Comment:
This shouldn't be necessary. The window already keeps a list of transactions that can be updated if necessary. The window could just needs to render them for the appropriate widget when needed. The transaction icon should be a single icon and a `widget.Button` rendered on the transaction to facilitate clicks.